lib/pages/proflile/create_pin/account_create_pin_page.dart

```diff
@@ -0,0 +1,193 @@
+  @override
+  State<AccountCreatePinPage> createState() => _AccountCreatePinPageState();
+}
+
+class _AccountCreatePinPageState extends State<AccountCreatePinPage>
+    with WidgetsBindingObserver {
+  bool isKeyboardVisible = false;
+
+  bool _isEnableBtn = false;
+  final pin = ["", "", "", ""];
+
+  void _updateBtnState() {
+    if (pin.length == 4) {
+      setState(() {
+        _isEnableBtn = pin.every((element) => element.isNotEmpty);
+      });
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    isKeyboardVisible = MediaQuery.of(context).viewInsets.bottom > 0;
+    return Scaffold(
+      backgroundColor: AppColors.black,
+      body: Column(
+        children: [
+          Expanded(
+            child: SingleChildScrollView(
+              child: Column(
+                children: [
+                  _buildLogo(),
+                  _buildTextDescription(),
+                  SizedBox(height: 48.h),
+                  _builImageProfile(),
+                  SizedBox(height: 24.h),
+                  _buildPinForm(),
+                  _buildTextInfo(),
+                  SizedBox(height: 20.h),
+                ],
+              ),
+            ),
+          ),
+          _buildBtn(context),
+        ],
+      ),
+    );
+  }
+
+  Container _buildBtn(BuildContext context) {
+    return Container(
+      alignment: Alignment.bottomCenter,
+      margin: EdgeInsets.only(
+          bottom: isKeyboardVisible ? 20.h : 80.h, left: 30.w, right: 30.w),
+      child: SecondaryButton(
+        backgroundColor:
+            _isEnableBtn ? AppColors.crimsonApprox : AppColors.black,
+        width: 1.sw,
+        height: 50.h,
+        title: context.l10n.iamAllSafeNow,
+        onPressed: _isEnableBtn
+            ? () => context.go(
+                    AccountPage.routeLocation +
+                        AccountCreateSuccessPage.routeLocation,
+                    extra: {
+                      "imagePath": widget.imagePath,
+                      "username": widget.username,
+                    })
+            : null,
+      ),
+    );
+  }
+
+  Container _buildTextInfo() {
+    return Container(
+      alignment: Alignment.center,
+      padding: EdgeInsets.only(left: 64.w, right: 64.w, top: 20.h),
+      child: Text(
+        context.l10n.createPinDescription,
+        style: GoogleFonts.inter(
+            fontSize: 16.sp,
+            fontWeight: FontWeight.w800,
+            color: AppColors.white.withOpacity(0.5)),
+        textAlign: TextAlign.center,
+      ),
+    );
+  }
+
+  Padding _buildPinForm() {
+    return Padding(
+      padding: EdgeInsets.symmetric(horizontal: 64.w),
+      child: Form(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: pin
+              .mapIndexed((index, element) => _buildPinInput(index))
+              .toList(),
+        ),
+      ),
+    );
+  }
```

Comment:
I think pin widget have complex logics, in this case we should extract it to specific class